src/Castle.Windsor/MicroKernel/Lifestyle/Scoped/ThreadStaticLifetimeScope.cs

```diff
@@ -12,9 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#if !FEATURE_REMOTING && !FEATURE_ASYNCLOCAL
 namespace Castle.MicroKernel.Lifestyle.Scoped
 {
-#if SILVERLIGHT
```

Comment:
In this case I would rather remove FEATURE_ASYNCLOCAL because there will be needed  `#if FEATURE_REMOTING || FEATURE_ASYNCLOCAL` in lots of places